Simple Incremental/Assets/Scripts/WeaponHook.cs

```diff
@@ -0,0 +1,33 @@
+﻿using UnityEditor.SceneManagement;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class WeaponHook : MonoBehaviour
+{
+    [SerializeField]
+    WeaponTemplate weaponTemplate = null;
+    ProjectileWeapon projectileWeapon = null;
+
+    public void Awake()
+    {
+        projectileWeapon = GetComponent<ProjectileWeapon>();
+    }
+
+    public bool Hook()
+    {
+        if (weaponTemplate != null)
+        {
+            if (projectileWeapon != null)
```

Comment:
You can require component for ProjectileWeapon so you don't have to check if its null.

Reply:
There will be other types of weapons. The design for handling selecting between them has yet to be determined, but since there may be other types of weapons I'm hesitant to limit it to a required component.